src/types.rs

```diff
@@ -401,13 +403,15 @@ impl Rewrite for ast::WherePredicate {
                     };
                     let used_width = type_str.len() + colon.len();
                     let budget = try_opt!(shape.width.checked_sub(used_width));
-                    let bounds_str: String = try_opt!(bounds
-                        .iter()
-                        .map(|ty_bound| {
-                            ty_bound
-                                .rewrite(context, Shape::legacy(budget, shape.indent + used_width))
-                        })
-                        .collect::<Option<Vec<_>>>())
+                    let bounds_str: String = try_opt!(
+                        bounds
+                            .iter()
+                            .map(|ty_bound| {
+                                ty_bound.rewrite(context,
+                                                 Shape::legacy(budget, shape.indent + used_width))
+                            })
+                            .collect::<Option<Vec<_>>>()
+                    )
```

Comment:
Thanks for pointing out, this was caused by rustfmt not allowing the chain item to stay on the same line with the closing parens when `context.use_block_indent()` returns `false`.
In the similar example, `.join(", ")` was kept on the same line because the chain was inside the function call which was formatted in block style and hence `context.use_block_indent()` returned `true`.